### PR TITLE
Group tools for planemo testing

### PR DESCRIPTION
**Pull request by bernt-matthias**
- simplifies the logic a bit since tools and data managers don't need to
be treated separately.
- check if every path in find_tools is in the output of find_repos, i.e.
check for missing .shed.yml

- all linters run now on the result of ci_find_repos
- tool testing uses ci_find_tools

as suggested here https://github.com/galaxyproject/tools-iuc/pull/3248#issuecomment-708993697

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bernt-matthias** [comment]
@nsoranzo .. I guess we should test this with some tool bumps or mock tools .. and suggestions which cases you would like to see?

**Comment by nsoranzo** [comment]
Can you rebase this PR please?

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -288,43 +303,24 @@ jobs:
...
-                               $(cat ../workflow_artifacts/changed_repositories.list)
-            else
-                planemo ci_find_repos --chunk_count ${{ needs.setup.outputs.nchunks }} --chunk ${{ matrix.chunk }} \
-                               --output changed_repositories_chunk.list \
-                               $(cat ../workflow_artifacts/changed_repositories.list)
-            fi
-        fi
-    - name: Show changed tools/repositories chunk list
-      run: cat changed_tools_chunk.list changed_repositories_chunk.list
+      run: planemo ci_find_tools --chunk_count ${{ needs.setup.outputs.nchunks }} --chunk ${{ matrix.chunk }} --exclude packages --exclude deprecated --exclude_from .tt_skip --output changed_tools_chunk.list $(cat ../workflow_artifacts/changed_repositories.list)
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:306]: While merging I removed `--changed_in_commit_range $COMMIT_RANGE ` .. wondering if this is needed

Comment (nsoranzo) [inline .github/workflows/pr.yaml:306]: We still need to check that `changed_repositories.list` is not empty:

```suggestion
      run: |
        touch changed_tools_chunk.list
        if [ -s ../workflow_artifacts/changed_repositories.list ]; then
          planemo ci_find_tools --chunk_count ${{ needs.setup.outputs.nchunks }} --chunk ${{ matrix.chunk }} --exclude packages --exclude deprecated --exclude_from .tt_skip --output changed_tools_chunk.list $(cat ../workflow_artifacts/changed_repositories.list)
        fi
```

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -288,43 +303,24 @@ jobs:
...
-                    TESTPATH="$DIR"
-                fi
-                PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy $PLANEMO_OPTIONS --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$DIR"/tool_test_output.json "$TESTPATH" || true
-                docker system prune --all --force --volumes || true
-            done < changed_repositories_chunk.list
-        else
-            echo '{"tests":[]}' > tool_test_output.json
+        mkdir json_output/
+        while read -r TOOLS; do
+           if grep -qf .tt_biocontainer_skip <(echo $TOOLS); then
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:313]:
```suggestion
           if grep -q "$TOOL" .tt_biocontainer_skip; then
```

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -288,43 +303,24 @@ jobs:
...
-                else
-                    TESTPATH="$DIR"
-                fi
-                PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy $PLANEMO_OPTIONS --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$DIR"/tool_test_output.json "$TESTPATH" || true
-                docker system prune --all --force --volumes || true
-            done < changed_repositories_chunk.list
-        else
-            echo '{"tests":[]}' > tool_test_output.json
+        mkdir json_output/
+        while read -r TOOLS; do
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:312]:
```suggestion
        while read -r TOOL; do
```

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -288,43 +303,24 @@ jobs:
...
+           if grep -qf .tt_biocontainer_skip <(echo $TOOLS); then
+               PLANEMO_OPTIONS=""
+           else
+               PLANEMO_OPTIONS="--biocontainers --no_dependency_resolution --no_conda_auto_init"
+           fi
+           json=$(mktemp -u -p json_output --suff .json)
+           PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json $json $TOOLS || true
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:319]:
```suggestion
           PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$json" "$TOOL" || true
```

File: .github/workflows/pr.yaml [hunk 5]
```diff
@@ -288,43 +303,24 @@ jobs:
...
-                    TESTPATH=$(planemo ci_find_tools "$DIR")
-                else
-                    TESTPATH="$DIR"
-                fi
-                PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy $PLANEMO_OPTIONS --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$DIR"/tool_test_output.json "$TESTPATH" || true
-                docker system prune --all --force --volumes || true
-            done < changed_repositories_chunk.list
-        else
-            echo '{"tests":[]}' > tool_test_output.json
+        mkdir json_output/
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:311]:
```suggestion
        mkdir json_output
```

File: .github/workflows/pr.yaml [hunk 6]
```diff
@@ -288,43 +303,24 @@ jobs:
...
+               PLANEMO_OPTIONS=""
+           else
+               PLANEMO_OPTIONS="--biocontainers --no_dependency_resolution --no_conda_auto_init"
+           fi
+           json=$(mktemp -u -p json_output --suff .json)
+           PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json $json $TOOLS || true
+           docker system prune --all --force --volumes || true
+        done < changed_tools_chunk.list
+        if [ ! -s changed_tools_chunk.list ]; then
+            echo '{"tests":[]}' > $(mktemp -u -p json_output --suff .json)
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:323]:
```suggestion
            echo '{"tests":[]}' > "$(mktemp -u -p json_output --suff .json)"
```

File: .github/workflows/pr.yaml [hunk 7]
```diff
@@ -289,42 +304,27 @@ jobs:
...
-            echo '{"tests":[]}' > tool_test_output.json
+        mkdir json_output
+        while read -r TOOL; do
+           if grep -q "$TOOL" .tt_biocontainer_skip; then
+               PLANEMO_OPTIONS=""
+           else
+               PLANEMO_OPTIONS="--biocontainers --no_dependency_resolution --no_conda_auto_init"
+           fi
+           json=$(mktemp -u -p json_output --suff .json)
+           PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$json" "$TOOL" || true
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:323]:
```suggestion
           PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy $PLANEMO_OPTIONS --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$json" "$TOOL" || true
```

File: .github/workflows/pr.yaml [hunk 8]
```diff
@@ -289,42 +304,27 @@ jobs:
...
+           docker system prune --all --force --volumes || true
+        done < changed_tools_chunk.list
+        if [ ! -s changed_tools_chunk.list ]; then
+            echo '{"tests":[]}' > "$(mktemp -u -p json_output --suff .json)"
         fi
     - name: Merge tool_test_output.json files
       run: find . -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:330]:
```suggestion
      run: planemo merge_test_reports json_output/*.json tool_test_output.json
```

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @bernt-matthias !

**Pull request merged**